src/DotVVM.Framework/ViewModel/Serialization/DefaultViewModelSerializer.cs

```diff
@@ -74,13 +76,20 @@ public void BuildViewModel(IDotvvmRequestContext context)
             {
                 throw new Exception($"Could not serialize viewModel of type { context.ViewModel.GetType().Name }. Serialization failed at property { writer.Path }. {GeneralViewModelRecommendations}", ex);
             }
+            var viewModelToken = writer.Token;
+
+            string viewModelCacheId = null;
+            if (context.Configuration.ExperimentalFeatures.ServerSideViewModelCache.IsEnabledForRoute(context.Route.RouteName))
+            {
+                viewModelCacheId = viewModelServerCache.StoreViewModel(context, (JObject)viewModelToken);
```

Comment:
One more thing, we should only store properties that sent to client **and also back to server**. This way we are wasting quite a bit of space. Fortunately, the serializer should silently ignore properties that should not be sent to server, so there is no change in behavior.

Unless you use the `Direction.ClientToServerNotInPostbackPath`, then the serializer will take it into account even though it should not be sent at all (assuming it was not in the path). Unfortunately, these properties can't be just dropped as they might also be needed when they are in the path. And, on the server side, we have basically no way of knowing which object are in the path during the serialization phase, so I don't see a simple fix to that :/ Maybe we could figure out the JSON path on client and send it to the server.